File: Interpolation/jl/methods.jl
function Lagrange(X, Y, x)
    if (length(X) != length(Y))
        println("Lagrange Interpolation: X and Y have different legnths.");
        return;
    end
    y = 0;
    n = length(X);
    for i = 1:n
        p = Y[i];
        for j = 1:n
            if (i != j)
                p *= (x - X[j])/(X[i] - X[j]);
            end
        end
        y += p;
    end
    return y;
end

File: Interpolation/cpp/methods.cc
#include "methods.h"

double Lagrange(double *X, double *Y, int n, double x) {
  double y = 0;

  for(int i = 0; i < n; i++) {
    double p = Y[i];
    for(int j = 0; j < n; j++) {
      if(j != i)
        p *= (x - X[j])/(X[i] - X[j]);
    }
    y += p;
  }

  return y;
}
